tinygrad/codegen/linearizer.py

```diff
@@ -223,6 +227,7 @@ def linearize(self):
       # TODO: the strides of this can be controlled
       self.sts.append(ShapeTracker(tuple([1] * self.first_reduce + self.group_for_reduce + [1] * (self.shape_len - self.upcasted - len(self.group_for_reduce) - self.first_reduce) + [x[0] for x in self.upcasted_axis(0)])))
       self.bufs.append(LocalBuffer("temp", self.sts[-1].size()))
+      self.bufmap.append(max(self.bufmap) + 1)
```

Comment:
Does a non realized buffer have to go in the bufmap?

Reply:
global_load and global_store are called with -1 to load LocalBuffer, and they need to ask self.bufmap to find the pos after dedup

Reply:
Ahh, why don't you just handle that case in load and store?

Reply:
The -1 thing is a stupid hack anyway...want to just make it the right thing?